customers/services/profile.py

```diff
@@ -127,13 +133,117 @@ def get_profile(self, id: UUID) -> HelsinkiProfileUser:
+                            primary_email: primaryEmail {{
+                                email
+                            }}
+                            primary_phone: primaryPhone {{
+                                phone
+                            }}
+                        }}
+                    }}
+                }}
+            }}
+        """
+        response = self.query(query)
+        profiles = (
+            response.get("profiles", {}).get("edges", [])
+            if response
+            and response.get("profiles", {})
+            and response.get("profiles", {}).get("edges")
+            else []
+        )
+
+        if force_only_one:
+            if len(profiles) > 1:
+                ids = [from_global_id(node.get("id")) for node in profiles]
+                raise MultipleProfilesException(ids=ids)
+            elif len(profiles) == 0:
+                raise NoProfilesException
+
+        users = []
+        for profile_edge in profiles:
+            users.append(self.parse_user_edge(profile_edge))
+
+        if force_only_one:
+            return users[0]
+
+        return users
+
+    @transaction.atomic
+    def create_profile(
+        self, first_name: str, last_name: str, email: str = None, phone: str = None
+    ):
+        from ..models import CustomerProfile
+
+        variables = {
+            "input": {"profile": {"firstName": first_name, "lastName": last_name}}
+        }
+        if email:
+            variables["input"]["profile"]["addEmails"] = [
+                {"primary": True, "email": email, "emailType": "NONE"}
+            ]
+        if phone:
+            variables["input"]["profile"]["addPhones"] = [
+                {"primary": True, "phone": phone, "phoneType": "NONE"}
+            ]
+
+        mutation = """
+        mutation CreateProfile($input: CreateProfileMutationInput!) {
+            createProfile(input: $input) {
+                profile {
+                    id
+                }
+            }
+        }
+        """
+        response = self.query(query=mutation, variables=variables)
+        profile = None
+        if global_id := response.get("createProfile", {}).get("profile", {}).get("id"):
+            profile_id = from_global_id(global_id)
+            profile = CustomerProfile.objects.create(id=profile_id)
+        return profile
+
     def parse_user_edge(self, gql_edge: Dict[str, dict]) -> HelsinkiProfileUser:
         return self.parse_user(gql_edge.get("node"))
 
     def parse_user(self, profile: Dict[str, dict]) -> HelsinkiProfileUser:
         user_id = from_global_id(profile.pop("id"))
-        email = (profile.pop("primary_email") or {}).get("email")
-        phone = (profile.pop("primary_phone") or {}).get("phone")
+        email = None
+        phone = None
+        if "primary_email" in profile:
+            primary_email = profile.pop("primary_email")
+            email = primary_email is not None and primary_email.get("email")
```

Comment:
if primary_email is None, then email == False, is that intended?

Reply:
Good catch.